### ISSUE TITLE
Group dynamodb tables in AWS-Console

### ISSUE THREAD

**Issue by GeorgeBellTMH** [aws-amplify/amplify-cli#4278]
Is there a way to use the dynamodb->tables->actions->create group functionality when pushing tables to AWS? Ideally we would group tables by application/environment by default...it can be a bit of pain to find tables when there are lots of environments/apps.

**Comment by SwaySway** [comment]
This is something we currently don't support, though this is something we can consider as an enhancement.

**Comment by ranman** [comment]
I think this will work if you automatically tag the tables with `DDBTableGroupKey-{yourvaluehere}`

Then create/upate an AWS resource group with the resource groups api:
https://docs.aws.amazon.com/ARG/latest/APIReference/API_CreateGroup.html

**Comment by cdac901** [comment]
+1

**Comment by bishonbopanna** [comment]
+1 : This feature will really help when you have multiple amplify envs - like dev, qe and prod.

**Comment by ranman** [comment]
> > I think this will work if you automatically tag the tables with `DDBTableGroupKey-{yourvaluehere}`
> > Then create/upate an AWS resource group with the resource groups api:
> > https://docs.aws.amazon.com/ARG/latest/APIReference/API_CreateGroup.html
> 
> @ranman I checked the link you provided and it does not speak about **`DDBTableGroupKey `**. Could you please tell me how can we use it?

Sure - if you do the following things with the resource groups api you'll see the table groups get auto created in the dynamodb console.

1. Create a group - use a tag query with the specific construction of `DDBTableGroupKey-YOURKEYHERE` and the value as `YOURKEYHERE`
2. Next, tag the relevant DDB tables with `DDBTableGroupKey-YOURKEYHERE` and the value `YOURKEYHERE`

Here's some example python code for creating the group:
```python
import json

import boto3

rg = boto3.client('resource-groups')
resource_query = json.dumps({
    "ResourceTypeFilters": ["AWS::DynamoDB::Table"], 
    "TagFilters": [
        {"Key": "DDBTableGroupKey-PROD", "Values": ["PROD"]}
    ]
})
rg.create_group(
    Name='PROD',
    ResourceQuery={'Type': 'TAG_FILTERS_1_0', 'Query': resource_query}
)
```

I think figuring out the tagging is the easier part.

**Comment by renebrandel** [comment]
Hi @GeorgeBellTMH - while we don't have DDB table group support, we do auto-tag all the tables based on your tags.json file. Would this satisfy your use case? https://aws.amazon.com/blogs/mobile/organizing-your-aws-resources-using-tags-with-the-amplify-cli/

**Comment by GeorgeBellTMH** [comment]
Yes I think so, it would be great if new projects were automatically tagged so as to create a group {projectName}-{env} by default if that is possible.
